### ISSUE TITLE
Issue in using  tf::Taskflow.Composed_of() functionality 

### ISSUE THREAD

**Issue by SundaramRathinam** [taskflow/taskflow#437]
**Describe the bug**
Trying to use Composition functionality to make my workflow modular. Facing issue in keeping target_Taskflow  alive.
To resolve the above issue
- Either i have to define n number of TaskFlow at global level , will be knowing at run time only which part of my workflow will be created as module
- Created map to store the n number of taskflow based on run time information
- There is no default constructor available in tf::Taskflow that handles this scenario
   - tf::Taskflow tmpTaskflow = taskflow1
   - I have to use std::move to overcome this, using std::move then the already created empty taskflow will be cleaned up

Create

**To Reproduce**
void CreateSubProcess(tf::Taskflow& taskflow,
                      tf::Task& parentTask,
                      tf::Task& subProcess)
{
    tf::Taskflow tmpTaskFlow("SubProcess");
    subProcess = taskflow.composed_of(tmpTaskFlow).name("SubProcess");
    subProcess.succeed(parentTask);

    tf::Task Start2 = tmpTaskFlow.emplace([&]() { std::cout << "Start2" << "\n"; }).name("Start2");

    tf::Task SubProcess_Task1 = tmpTaskFlow.emplace([&]() { std::cout << "SubProcess_Task1" << "\n"; }).name("SubProcess_Task1");
    SubProcess_Task1.succeed(Start2);

    tf::Task SubProcess_Task2 = tmpTaskFlow.emplace([&]() { std::cout << "SubProcess_Task2" << "\n"; }).name("SubProcess_Task2");
    SubProcess_Task2.succeed(SubProcess_Task1);

    tf::Task End2 = tmpTaskFlow.emplace([&]() { std::cout << "End2" << "\n"; }).name("End2");
    End2.succeed(SubProcess_Task2);
}

int main()
{
    tf::Taskflow taskflow;
    tf::Task task1 = taskflow.emplace([&]() { std::cout << "Start1" << "\n"; }).name("Start1");
    tf::Task task2 = taskflow.emplace([&]() { std::cout << "Task1" << "\n"; }).name("Task1");
    task2.succeed(task1);

    tf::Task subProcess;
    CreateSubProcess(taskflow, task2, subProcess);

    tf::Task task4 = taskflow.emplace([&]() { std::cout << "End1" << "\n"; }).name("End");
    task4.succeed(subProcess);
    taskflow.dump(std::cout);

    tf::Executor executor;
    executor.run(taskflow).wait();
    return 0;
}
Its dumps in taskflow.dump as target taskflow is not alive

Example 2:
        std::vector<FCNODE> vSubProcess;
        GetSubProcess(pRoot, vSubProcess);
        std::map<FCNODE, tf::Taskflow&&> mSubProcessTaskFlow;
        for (int idxSubProcess = 0; idxSubProcess < vSubProcess.size(); idxSubProcess++)
        {
            FCNODE subProcess = vSubProcess[idxSubProcess];
            GuString sCaption = subProcess->GetNodeCaption();
            tf::Taskflow subProcessTaskFlow(sCaption.GetAnsiBuffer());
            mSubProcessTaskFlow.insert_or_assign(subProcess, subProcessTaskFlow);
        }

Will be passing mSubProcessTaskFlow to another function where this task flow will be populated with task
and the modues will be added to main taskflow

Compilation Error:
Severity	Code	Description	File	Project	Line	Suppression State
Error	C2280	'std::pair<const FCNODE,tf::Taskflow &&>::pair(const std::pair<const FCNODE,tf::Taskflow &&> &)': attempting to reference a deleted function	C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.26.28801\include\xmemory	VWorkflowExecuteService	671	

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

**Comment by tsung-wei-huang** [comment]
Hi, @SundaramRathinam , thanks for your questions. Your understanding is correct. Composing a taskflow from another taskflow must ensure both taskflow remain alive during the execution. `composed_of` does not own any taskflow but creates a soft linking (e.g., pointer) to the graph to compose.

Does this answer your question? Sorry, I saw quite a lot of messy code but I tried the best to understand them.
